suite/spdz2k/src/main/java/dk/alexandra/fresco/suite/spdz2k/protocols/computations/Spdz2kMacCheckComputation.java

```diff
@@ -74,7 +76,7 @@ public Spdz2kMacCheckComputation(Pair<List<Spdz2kSInt<PlainT>>, List<PlainT>> to
             List<byte[]> sharesLowBits = authenticatedElements.stream()
                 .map(element -> element.getShare().getLeastSignificant().toByteArray())
                 .collect(Collectors.toList());
-            return new BroadcastComputation<ProtocolBuilderNumeric>(sharesLowBits)
+            return new BroadcastComputation<ProtocolBuilderNumeric>(sharesLowBits, true)
```

Comment:
Maybe I am missing something but won't we also need to sample the coefficients for the MAC check in SPDZ2k as for SPDZ during the computation of the MAC check? In that case it seems that it is not enough to simply call sampleCoefficients in the constructor but it should also be part of the buildComputation.